parquet-hadoop/src/main/java/org/apache/parquet/hadoop/example/ValidatingUnsignedIntegerRecordConsumer.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.hadoop.example;
+
+import java.util.Stack;
+import org.apache.parquet.io.InvalidRecordException;
+import org.apache.parquet.io.api.RecordConsumer;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+
+/**
+ * A RecordConsumer that validates unsigned integer values to ensure they comply
+ * with the Parquet specification.
+ *
+ * This consumer wraps another RecordConsumer and validates that values written
+ * to unsigned integer fields (UINT_8, UINT_16, UINT_32, UINT_64) are within
+ * the valid unsigned range before delegating to the wrapped consumer.
+ */
+public class ValidatingUnsignedIntegerRecordConsumer extends RecordConsumer {
```

Comment:
I'm not sure if it is enough to modify lines below to add the check:

https://github.com/apache/parquet-java/blob/master/parquet-column/src/main/java/org/apache/parquet/io/ValidatingRecordConsumer.java#L203-L206

WDYT? @parthchandra